### PR TITLE
make `getData` return a reference to the `BytesData` of `Bytes` in PHP

### PR DESCRIPTION
**Pull request by mockey**

### CODE REVIEW

File: std/haxe/io/Bytes.hx [hunk 1]
```diff
@@ -448,8 +448,13 @@ class Bytes {
 		return s.toString();
 	}
 
-	public inline function getData() : BytesData {
+	#if php @:extern #end public inline function getData() : BytesData {
+		#if php
+		var r = this;
```
Comment (Simn) [inline std/haxe/io/Bytes.hx]: I guess this can go too now?

Comment (mockey) [inline std/haxe/io/Bytes.hx]: Sure, I missed that. The `@:extern` is still necessary, though, because `return &$this` is no valid PHP.

**Comment by Simn** [comment]
"Fatal error: Call-time pass-by-reference has been removed; If you would like to pass argument by reference, modify the declaration of _hx_string_or_null(). in /Users/travis/build/HaxeFoundation/haxe/tests/unit/bin/php/lib/unit/TestSpecification.class.php on line 1487"

:(

**Comment by mockey** [comment]
Yeah, doesn't work in functions. Should have known...

**Pull request closed without merging**